### ISSUE TITLE
Relative date selector improvements

### ISSUE THREAD

**Issue by maliayas** [metabase/metabase#2142]
-  My Metabase version is 0.15.0

I propose to modify relative date selector like below:

![ekran goruntusu 2016 03 11 17_51_52](https://cloud.githubusercontent.com/assets/844008/13707754/d7cf52b0-e7b3-11e5-8c44-ac4ea7878009.png)

Changed parts:
- Added PAST header and removed "Past 7 days", "Past 30 days" buttons as they have a special line now.
- Switched THIS and LAST lines to be more logical in the new case.

**Comment by MrMauricioLeite** [comment]
An option to choose _____ days would be awesome too.
So some guys here like to see data from last 90 days or last 12 weeks. It
sure is a flexible enough way of filtering.

Also, I miss some way of filtering with >= or even <=. For example I want
to see sum of revenue since 2016-01-18 (a day we made a significant change
in the UI of our website).


[]'s,

Mauricio R. Leite
(11) 9 4141  2201

2016-03-11 13:08 GMT-03:00 Ali Ayas <notifications@github.com>:

**Comment by robinfrick** [comment]
@MrMauricioLeite that's possible. There are small arrows in the specific date view that appear once you select a date which do exactly what you want.
But it also took me more than a month to notice them. Maybe they should be more obvious. For me they look a lot like I can go to the next or previous month rather than <= / >=
![image](https://cloud.githubusercontent.com/assets/8144392/13782092/022c4b52-eac6-11e5-9242-bcc0e7ab21a4.png)

**Comment by MrMauricioLeite** [comment]
Wow, I woul never guess this.

Cant we turn those buttons into big purple buttons above the calendar?
What do you think @agilliland ??

**Comment by mazameli** [comment]
Good feedback, guys. Thanks. (cc @kdoh)

**Comment by kdoh** [comment]
Hey all. Like @mazameli said this feedback is incredibly valuable. In addition to improving the discoverability and usability of what's already there, we want to make sure we're including an appropriate range of time based filtering options so all this is great.

If you can think any other examples of common types of work, or common ranges you find yourself using with date based filters it'd be great to keep expanding this list here.

To summarize what's been mentioned so far it sounds like:
- More and / or custom intervals to allow for a greater set of relative range needs
- Better discoverability of before / after <date>

One other thing I'll throw out there that would be nice to try and explore is the ability to filter by relative future dates as well. That would unlock the ability to ask questions like "Show me all my reservations next month" or "next year."

**Comment by anthonysexton** [comment]
If you are looking for date picker inspiration, this is one of the best ones I've come across:

http://baremetrics.github.io/calendar/

Although it may not tick all of the flexibility boxes relative to the current one.

**Comment by kdoh** [comment]
Thanks for the link @anthonysexton! I'll add it to the inspiration pile.

**Comment by maliayas** [comment]
I'd like to ask you guys which one you think is the more common industry need: including "today" in "past 7 days" and "past 30 days" or not? Personally, I need it to be included, so that I can see today's performance in the last 30 days' chart. What do you think?

Also do you think a "last 24 hours" would be fine to have as an option in the relative date selector?

P.S. I may sound like a contributor but I'm not. I just wanted to discuss before suggesting.

**Comment by key88sf** [comment]
I'm also +1 for including "today" in the "Past N days" selections (or at least having an option to include it). We often want reports to include today's data.

**Comment by agilliland** [comment]
If this was to be done I'd say it definitely needs to be a configuration choice.  It's certainly not appropriate to force the current (and partial) day of data to be included in queries in a lot of cases.

**Comment by maliayas** [comment]
It may be irrelevant in reports but it makes sense to have in dashboards. A checkbox in the same box might solve the issue for both cases.

**Comment by key88sf** [comment]
I agree it should be a checkbox (or similar) type of option. I'd prefer it on the question-level as opposed to the dashboard-level.

**Comment by maliayas** [comment]
> as opposed to the dashboard-level

My suggestion is date selector box level.

**Comment by salsakran** [comment]
closing this as we covered everything except the question of whether "past 7" includes today in #3750
